src/main/java/config/MyApplicationContextConfiguration.java

```diff
@@ -5,42 +5,73 @@
 import org.postgresql.ds.PGSimpleDataSource;
 
 import javax.sql.DataSource;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.ComponentScan;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Scope;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.*;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.transaction.PlatformTransactionManager;
+import service.TicketService;
+import service.UserService;
 
 import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.Properties;
 
 @Configuration
 @ComponentScan
+@PropertySource("classpath:application.properties")
 public class MyApplicationContextConfiguration {
 
+    @Value("${url}")
+    private String url;
+
+    @Value("${login}")
+    private String login;
+
+    @Value("${password}")
+    private String password;
+
     private static File file = new File("src/main/resources/ConnectionConfiguration.txt");
-    private static ConfigurationObject configurationObject = ConfigurationReader.read(file);
+
+    @Bean
+    public PlatformTransactionManager txManager() {
+        return new DataSourceTransactionManager(dataSource());
+    }
 
     @Bean
     @Scope("singleton")
     public DataSource dataSource() {
         PGSimpleDataSource pgSimpleDataSource = new PGSimpleDataSource();
-        pgSimpleDataSource.setUser(configurationObject.getLogin());
-        pgSimpleDataSource.setPassword(configurationObject.getPassword());
-        pgSimpleDataSource.setURL(configurationObject.getUrl());
+        pgSimpleDataSource.setUser(login);
+        pgSimpleDataSource.setPassword(password);
+        pgSimpleDataSource.setURL(url);
         return pgSimpleDataSource;
     }
 
     @Bean
-    public UserDAO userDao() {
-        return new UserDAO(dataSource());
+    public JdbcTemplate jdbcTemplate() {
+        return new JdbcTemplate(dataSource());
+    }
+
+    @Bean
+    public UserService userService() {
+        return new UserService();
+    }
+
+    @Bean
+    public UserDAO userDAO() {
+        return new UserDAO(dataSource(), jdbcTemplate());
+    }
+
+    @Bean
+    public TicketService ticketService() {
+        return new TicketService();
     }
```

Comment:
When you define bean in configuration class - it is bean.
When annotate class with `@Component`, `@Repository`, `@Service` or `@Controller` - it is also  bean.

There is no need to use both approaches. Choose just 1 for each object and remove redundant.

Reply:
Fixed. I chose to define the Service bean in the configuration class. And I'm wonder, why if I am trying to define only using annotations @Component / @Service in the service class, I get the NoSuchBeanDefinitionexception?

![No qualifying bean of type](https://github.com/user-attachments/assets/08913bd8-435b-4caf-85a0-3c0c999c72dd)

Reply:
it happens because you use `@ComponentScan` in your configuration class without arguments. By default Spring scans only this package and subpackages.
In order to fix this behavior, you need to move all classes and packages from root folder `src/main/java` into specific package and is as argument to the `@ComponentScan`:

1. move all from `src/main/java` to `src/main/java/com/project/jfb`
2. specify argument: `@ComponentScan("com.project.jfb")`

Reply:
And if you decided to create bean through the Configuration file, then the injection of beans have to be also in this place, not through the field's injection in service class.
Use:
```java
    @Bean
    public ApplicationService applicationService(
        UserDAO userDAO,
        TicketDAO ticketDAO,
        @Value("${isUsersUpdateAndTicketsCreationAvailable}") boolean isUsersUpdateAndTicketsCreationAvailable) {
        return new ApplicationService(userDAO, ticketDAO, isUsersUpdateAndTicketsCreationAvailable);
    }
```
and
```java
@AllArgsConstructor
public class ApplicationService implements Printable {
    private final UserDAO userDAO;
    private final TicketDAO ticketDAO;
    private boolean isUsersUpdateAndTicketsCreationAvailable;
```

instead of:
```java
    @Bean
    public ApplicationService applicationService() {
        return new ApplicationService();
    }
```
and
```java
    @Autowired
    private UserDAO userDAO;

    @Autowired
    private TicketDAO ticketDAO;

    @Value("${isUsersUpdateAndTicketsCreationAvailable}")
    private boolean isUsersUpdateAndTicketsCreationAvailable;
```

Reply:
Fixed. Thank you